diff --git a/CMakeLists.txt b/CMakeLists.txt
index d39d68f5..68d627a9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,8 @@ set (project_cmake_dir ${PROJECT_SOURCE_DIR}/cmake
 
 message (STATUS "${PROJECT_NAME} version ${SDF_VERSION_FULL}")
 set (CMAKE_INCLUDE_DIRECTORIES_PROJECT_BEFORE ON)
+set (CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES "")
+set (CMAKE_CXX_IMPLICIT_LINK_DIRECTORIES "")
 
 #============================================================================
 # We turn off extensions because (1) we do not ever want to use non-standard
@@ -53,6 +55,8 @@ set (CMAKE_INCLUDE_DIRECTORIES_PROJECT_BEFORE ON)
 # choose -std=c++14.
 # See https://github.com/ignitionrobotics/ign-cmake/issues/13 for more info.
 set(CMAKE_CXX_EXTENSIONS off)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # Include GNUInstallDirs to get canonical paths
 include(GNUInstallDirs)
diff --git a/cmake/SDFUtils.cmake b/cmake/SDFUtils.cmake
index 525c4d4c..6f7b23f2 100644
--- a/cmake/SDFUtils.cmake
+++ b/cmake/SDFUtils.cmake
@@ -154,6 +154,7 @@ macro (sdf_build_tests)
         ${sdf_target}
         pthread
         ${IGNITION-MATH_LIBRARIES}
+        PkgConfig::URDF
       )
     elseif(WIN32)
       target_link_libraries(${BINARY_NAME} PRIVATE
diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index a6c66195..6908c68f 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -28,7 +28,7 @@ endif()
 
 if (NOT DEFINED USE_INTERNAL_URDF OR NOT USE_INTERNAL_URDF)
   # check for urdfdom with pkg-config
-  pkg_check_modules(URDF urdfdom>=1.0)
+  pkg_check_modules(URDF urdfdom>=1.0 IMPORTED_TARGET)
 
   if (NOT URDF_FOUND)
     find_package(urdfdom)
@@ -43,10 +43,6 @@ if (NOT DEFINED USE_INTERNAL_URDF OR NOT USE_INTERNAL_URDF)
     else()
       BUILD_ERROR("Couldn't find the urdfdom >= 1.0 system installation")
     endif()
-  else()
-    # what am I doing here? pkg-config and cmake
-    set(URDF_INCLUDE_DIRS ${URDF_INCLUDEDIR})
-    set(URDF_LIBRARY_DIRS ${URDF_LIBDIR})
   endif()
 endif()
 
