diff --git a/src/Utils_TEST.cc b/src/Utils_TEST.cc
index d06271fd..d5fa3ba7 100644
--- a/src/Utils_TEST.cc
+++ b/src/Utils_TEST.cc
@@ -52,21 +52,6 @@ TEST(DOMUtils, PoseNoFrame)
   EXPECT_TRUE(frame.empty());
 }
 
-/////////////////////////////////////////////////
-TEST(DOMUtils, PoseNoValue)
-{
-  sdf::ElementPtr element(new sdf::Element);
-  element->SetName("pose");
-  element->AddValue("pose", "", true);
-
-  ignition::math::Pose3d pose;
-  std::string frame;
-  EXPECT_TRUE(sdf::loadPose(element, pose, frame));
-
-  EXPECT_EQ(ignition::math::Pose3d::Zero, pose);
-  EXPECT_TRUE(frame.empty());
-}
-
 /////////////////////////////////////////////////
 TEST(DOMUtils, PoseWithFrame)
 {
